kafka/event.go

```diff
@@ -231,6 +231,18 @@ out:
 				setupTopicPartitionFromCrktpar((*TopicPartition)(&peof), crktpar)
 
 				retval = peof
+			case C.RD_KAFKA_RESP_ERR__FATAL:
+				// A fatal error has been raised.
+				// Extract the actual error from the client
+				// instance and return a new Error with
+				// fatal set to true.
+				cFatalErrstrSize := C.size_t(512)
+				cFatalErrstr := (*C.char)(C.malloc(cFatalErrstrSize))
+				defer C.free(unsafe.Pointer(cFatalErrstr))
+				cFatalErr := C.rd_kafka_fatal_error(h.rk, cFatalErrstr, cFatalErrstrSize)
+				fatalErr := newErrorFromCString(cFatalErr, cFatalErrstr)
+				fatalErr.fatal = true
```

Comment:
newErrorFromCString is not exported anyway so why not add a isFatal argument and use that? Alternatively you could use and enum for log severity to make it more robust.

Reply:
I didn't want to change the 10 other places that don't need to set the fatal thing.